controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -539,6 +545,26 @@ func getCurrentImage(host *metal3v1alpha1.BareMetalHost) *metal3v1alpha1.Image {
 	return nil
 }
 
+// actionDetached() detaches the host from the Provisioner
+func (r *BareMetalHostReconciler) actionDetached(prov provisioner.Provisioner, info *reconcileInfo) actionResult {
+	provResult, err := prov.Detach()
+	if err != nil {
+		return actionError{errors.Wrap(err, "failed to detach")}
+	}
+	if provResult.ErrorMessage != "" {
+		return recordActionFailure(info, metal3v1alpha1.DetachError, provResult.ErrorMessage)
+	}
+	if provResult.Dirty {
```

Comment:
If Dirty is true then we want to clear any DetachError (but not set the operational status or reset the ErrorCount).

Reply:
Ok adjusted to call clearError like in similar provResult.Dirty cases.